[CI] StreamingParallelParsingCoordinatorTests testCloseWhileSegmentatorParkedOnDispatchPermit failing

**Build Scans:**
- [elasticsearch-periodic #12227 / openjdk21_checkpart3_java-fips-140-3-matrix](https://gradle-enterprise.elastic.co/s/uy34nnpl6fzbm)
- [elasticsearch-periodic #12227 / openjdk25_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/6m4mhpfv7kcia)
- [elasticsearch-pull-request #142676 / part-3](https://gradle-enterprise.elastic.co/s/qhxxpddcze7es)
- [elasticsearch-pull-request #142660 / part-3](https://gradle-enterprise.elastic.co/s/q63b3jmhkpblu)
- [elasticsearch-pull-request #142650 / part-3](https://gradle-enterprise.elastic.co/s/tj6w5wypwv77e)
- [elasticsearch-pull-request #142622 / part-3](https://gradle-enterprise.elastic.co/s/h7h5hjucv3x7m)
- [elasticsearch-intake #38996 / part3](https://gradle-enterprise.elastic.co/s/l2jw66ussb33i)
- [elasticsearch-pull-request #142552 / part-3](https://gradle-enterprise.elastic.co/s/462bvekouytgy)
- [elasticsearch-pull-request #142550 / part-3](https://gradle-enterprise.elastic.co/s/uabbvf2yajkws)
- [elasticsearch-pull-request #142541 / part-3](https://gradle-enterprise.elastic.co/s/znvxwtvnftb5i)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.datasources.StreamingParallelParsingCoordinatorTests.testCloseWhileSegmentatorParkedOnDispatchPermit" -Dtests.seed=F0F906A5D922759F -Dtests.locale=pa-Arab-PK -Dtests.timezone=America/Antigua -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.datasources.StreamingParallelParsingCoordinatorTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCloseWhileSegmentatorParkedOnDispatchPermit'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: close() must return within 5s of segmentator being parked
```

**Issue Reasons:**
- [main] 10 failures in test testCloseWhileSegmentatorParkedOnDispatchPermit (7.4% fail rate in 135 executions)
- [main] 7 failures in step part-3 (8.9% fail rate in 79 executions)
- [main] 7 failures in pipeline elasticsearch-pull-request (9.7% fail rate in 72 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -396,9 +396,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.cache.SharedBlobCacheWarmingServiceIT
   method: testCacheIsWarmedBeforeIndexingShardRelocation_AfterHandoff
   issue: https://github.com/elastic/elasticsearch/issues/149177
-- class: org.elasticsearch.xpack.esql.datasources.StreamingParallelParsingCoordinatorTests
-  method: testCloseWhileSegmentatorParkedOnDispatchPermit
-  issue: https://github.com/elastic/elasticsearch/issues/148229
 - class: org.elasticsearch.xpack.stateless.StatelessHollowIndexShardsIT
   method: testStressWithoutFailures
   issue: https://github.com/elastic/elasticsearch/issues/148310

